packages/mongodb-chatbot-ui/src/useConversation.tsx

```diff
@@ -591,6 +595,22 @@ export function useConversation(params: UseConversationParams) {
     dispatch({ type: "setMessageMetadata", messageId, metadata });
   };
 
+  const updateMessageMetadata: ConversationActor["updateMessageMetadata"] =
+    async ({ messageId, metadata }) => {
+      if (!state.conversationId) {
+        console.error(`Cannot updateMessageMetadata without a conversationId`);
+        return;
+      }
+      dispatch({
+        type: "setMessageMetadata",
+        messageId,
+        metadata: {
+          ...state.messages.find((m) => m.id === messageId)?.metadata,
+          ...metadata,
```

Comment:
cna metadata be undefined? if so, i think prefer nullish spreading

```suggestion
          ...(metadata ?? {}),
```

Reply:
We validate that the event is well-formed & the field exists in the `addMessageStreaming` implementation so this won't be undefined. Also spread operator supports undefined.